From 57dc29ed954a4c51677355e8f147ccb9e0902e2c Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Mon, 8 Apr 2024 22:18:34 +0800
Subject: [PATCH] drm: img-rogue: drop DRIVER_MODESET capability

This will lead this GPU to be wrongly considered as a display device in
KWin.

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 .../gpu/drm/img/img-rogue/services/server/env/linux/pvr_drm.c   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/img/img-rogue/services/server/env/linux/pvr_drm.c b/drivers/gpu/drm/img/img-rogue/services/server/env/linux/pvr_drm.c
index 70391bd8e..b53924e3b 100644
--- a/drivers/gpu/drm/img/img-rogue/services/server/env/linux/pvr_drm.c
+++ b/drivers/gpu/drm/img/img-rogue/services/server/env/linux/pvr_drm.c
@@ -451,7 +451,7 @@ const struct file_operations pvr_drm_fops = {
 };
 
 const struct drm_driver pvr_drm_generic_driver = {
-	.driver_features	= DRIVER_MODESET | DRIVER_RENDER,
+	.driver_features	= DRIVER_RENDER,
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 18, 0))
 	.load			= NULL,
-- 
2.46.0

